flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReaderTest.java

```diff
@@ -115,6 +130,76 @@ public void testWakeUp() throws Exception {
         assertNull(error.get());
     }
 
+    @Test
+    public void testNumBytesInCounter() throws Exception {
+        // Create a task metric group for intercepting numBytesOut
+        final InterceptingTaskMetricGroup taskMetricGroup = new InterceptingTaskMetricGroup();
+        final OperatorMetricGroup operatorMetricGroup =
+                taskMetricGroup.getOrAddOperator(new OperatorID(0, 0), "fakeOperator");
+        KafkaPartitionSplitReader<Integer> reader =
+                createReader(
+                        new Properties(),
+                        InternalSourceReaderMetricGroup.wrap(operatorMetricGroup));
```

Comment:
I think it is better to introduce a new method `InternalSourceReaderMetricGroup#mock(MetricGroup metricGroup, OperatorIOMetricGroup opMetricGroup)` like I did here https://github.com/fapaul/flink/commit/01de3e6db4c68d316d02bda35279e163673aa2bc and not use the task metric.

Reply:
What about using Mockito? 

```java
// Mock an operator metric group for intercepting numBytesIn
final OperatorMetricGroup operatorMetricGroup =
        Mockito.mock(OperatorMetricGroup.class, Mockito.RETURNS_DEEP_STUBS);
Counter numBytesInCounter = new SimpleCounter();
Mockito.when(operatorMetricGroup.getIOMetricGroup().getNumBytesInCounter())
        .thenReturn(numBytesInCounter);
SourceReaderMetricGroup sourceReaderMetricGroup =
        InternalSourceReaderMetricGroup.wrap(operatorMetricGroup);
```

This can avoid implementing another ```NumBytesInOperatorIOMetricGroup``` with a bunch of ```return null``` stuff.

Reply:
Mockito is banned for new stuff since we got burnt several times in the past.
#16875 now managed to just use the existing stuff, so please check if you can use that.